### PR TITLE
Fix readme example

### PR DESCRIPTION
**Pull request by dgduncan**
Update the example in the README to better reflect real world use of the server. This includes adding signals, creating a go routine for `server.Serve()` and blocking until interrupted.

### CODE REVIEW

**Review by mochi-co — DISMISSED** [review]
This is a great idea, it more accurately represents real world usage and will help people get up and running easier. Approved! :)

File: README.md [hunk 1]
```diff
@@ -103,10 +112,18 @@ func main() {
     log.Fatal(err)
   }
   
-  err = server.Serve()
-  if err != nil {
-    log.Fatal(err)
-  }
+
+  go func() {
```
Comment (torntrousers) [inline README.md:116]: The server.Serve() doesn't actually need to be Go routine, does it? I suppose it doesn't matter much either way, but for the sake of keeping the readme example simple?

File: README.md [hunk 2]
```diff
@@ -90,6 +90,15 @@ import (
...
+  sigs := make(chan os.Signal, 1)
+  done := make(chan bool, 1)
+  signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+  go func() {
+    <-sigs
+    done <- true
+  }()
+
   // Create the new MQTT Server.
   server := mqtt.New(nil)
```
Comment (torntrousers) [inline README.md:103]: To get this to work I had to change the import to be ```mqtt "github.com/mochi-mqtt/server/v2"```

File: README.md (hunk 1 above)
Comment (dgduncan) [reply README.md:116]: It technically does not; however, in many of code [examples](https://github.com/mochi-mqtt/server/tree/fix-readme-example/examples) it is done with a go routine so I did this to remain consistent.

File: README.md (hunk 2 above)
Comment (dgduncan) [reply README.md:103]: Very good catch! I had to do the same thing but I forgot to copy over imports.

File: README.md (hunk 1 above)
Comment (dgduncan) [reply README.md:116]: I have no strong opinion either way. Perhaps we could wait to hear from @mochi-co to see if there was an original intent behind using the new go routine? I could see a world in which that was left over from a time when `.Serve()` did block which made it more necessary for the new routine.

**Review by torntrousers — APPROVED** [review]
That change to all the examples could be another PR though and this one LGTM

**Review by mochi-co — APPROVED** [review]

**Pull request merged**